sale_order_pricelist_tax/models/sale_order.py

```diff
@@ -19,6 +19,45 @@ def update_prices(self):
                 SaleOrder, record.with_context(pricelist=record.pricelist_id.id)
             ).update_prices()
 
+    price_include_tax_exception = fields.Html(
+        readonly=True, compute="_compute_price_include_tax"
+    )
+
+    price_include_tax = fields.Boolean(
+        compute="_compute_price_include_tax",
+    )
+
+    @api.depends("order_line", "order_line.tax_id", "order_line.tax_id.price_include")
+    def _compute_price_include_tax(self):
+        message = _(
+            """
+        Sale Order lines must have the same type of tax:
+        <br /> - Either price include tax
+        <br /> - Either price exclude tax
+        """
+        )
+        for rec in self:
```

Comment:
```
for rec in self:
    taxes = self._get_all_taxes()
    taxe_price_inc = set(taxes.mapped("price_include")
    if len(taxe_price_inc) > 1
       rec.price_tax_state = "exception"
    elif len(taxe_price_inc) == 1 and taxe_price_inc.pop():
       rec.price_tax_state = "include"
    else:
       rec.price_tax_state = "exclude"
```

Reply:
I put it in a mixin model to avoid duplicated code. 
@sebastienbeau , @bguillot